include/fmt/compile.h

```diff
@@ -141,7 +141,7 @@ class format_string_compiler : public error_handler {
     auto it = parse_format_specs(begin, end, handler);
     if (*it != '}') on_error("missing '}' in format string");
     repl.arg_id = part_.part_kind == part::kind::arg_index
-                      ? arg_ref<Char>(part_.val.arg_index)
+                      ? arg_ref<Char>(static_cast<int>(part_.val.arg_index))
```

Comment:
Please change the type of `arg_index` in `arg_ref` to `int` and remove the cast.

Reply:
It seems to be already 'int' in struct format_part.value . Will remove the cast.